packages/core/src/browser/preferences/preference-contribution.ts

```diff
@@ -66,11 +66,10 @@ export class PreferenceSchemaProvider extends PreferenceProvider {
     protected readonly preferences: { [name: string]: any } = {};
     protected validateFunction: Ajv.ValidateFunction;
 
-    constructor(
-        @inject(ContributionProvider) @named(PreferenceContribution)
-        protected readonly preferenceContributions: ContributionProvider<PreferenceContribution>
-    ) {
-        super();
+    @inject(ContributionProvider) @named(PreferenceContribution)
```

Comment:
why?

Reply:
Since the constructor would become trivial (just a call to super()), isn't it equivalent (and simpler) to inject the field directly and not have a constructor?

Reply:
ok, one should use `postConstruct` to hook into instantiation after injection